docs/molgenis/dev_graphql.md

```diff
@@ -1,25 +1,283 @@
-this will open up a GraphQL playground for the selected schema (or in case of 'central' the database level API. <br>
+* https://emx2.dev.molgenis.org/api/graphql - root api
+* https://emx2.dev.molgenis.org/pet%20store/api/graphql - api for database 'pet store'
 
-For example:
-* Database level API at https://emx2.dev.molgenis.org/api/graphql 
-* Databaswe level playground at https://emx2.dev.molgenis.org/apps/graphql-playground
-* Schema level API for 'pet store' at https://emx2.dev.molgenis.org/pet store/graphql 
-* Schema level playground for 'pet store' at https://emx2.dev.molgenis.org/pet store/graphql-playground 
+Full documentation can be found while visiting the graphql-playground app. You can click 'docs' there.
+
+* https://emx2.dev.molgenis.org/apps/graphql-playground/ - playground for 'root' api
+* https://emx2.dev.molgenis.org/pet%20store/graphql-playground/ - example for 'pet store' database
+
+## Functions available on all APIs.
+
+These functionalities are available for both the 'root' api and the database api.
+
+### Sign in
+
+Sign into an existing account. When running in a website then a session will be created. However, when you use a script
+you can retrieve a token to authenticate in other calls. Provide these in subsequent request headers as '
+x-molgenis-token'.
+
+```graphql
+mutation {
+    signin(password: "bofke", email: "bofke@some.org") {
+        message
+        token
+    }
+}
+```
+
+### Sign up
+
+```graphql
+mutation {
+    signup(email: "bofke@some.org", password: "somepass") {
+        message
+    }
+}
+```
+
+### Sign out
+
+```graphql
+mutation {
+    signout {
+        message
+    }
+}
+```
+
+### changePassword
+
+```graphql
+mutation {
+    changePassword(password: "password")
+}
+
+```
+
+### retrieve session object
+
+In the session object you can get active account ('email'), the roles of this user, and a listing of the schema's in the
+system.
+
+```graphql
+{
+    _session {
+        email
+        roles
+        schemas
+    }
+}
+```
+
+### retrieve settings
+
+MOLGENIS has a generic key/value settings for storing settings. Settings can be set on level of whole system, and per
+database (per database includes the overal settings).
+
+```graphql
+{
+    _settings {
+        key
+        value
+    }
+}
+```
+
+## Functions available for each database
+
+### query schema
+
+```graphql
+{
+    _schema {
+        name
+        tables {
+            name
+            id
+            externalSchema
+            inherit
+            tableType
+            description
+            columns {
+                name
+                id
+                description
+                position
+                columnType
+                key
+                required
+                refTable
+                refLink
+                refBack
+                refLabel
+                validation
+                visible
+                semantics
+            }
+            settings {
```

Comment:
how is setting nested within table ?

Reply:
what is the question? It a table setting (analogous to schema setting, column setting, user settings and database settings. We can have setting everywhere :-)

Reply:
I added documentation